runner/src/qemu.rs

```diff
@@ -0,0 +1,33 @@
+use super::Cli;
+use std::path::PathBuf;
+use std::process::{Child, Command, Stdio};
+
+// Spawns a QEMU VM with a simulated Tock system and the process binary. Returns
+// the handle for the spawned QEMU process.
+pub fn deploy(cli: &Cli, tab_path: PathBuf) -> Child {
+    let device = format!(
+        "loader,file={},addr=0x20040000",
+        tab_path
+            .into_os_string()
+            .into_string()
+            .expect("Non-UTF-8 path")
+    );
+    let mut qemu = Command::new("tock2/tools/qemu/build/qemu-system-riscv32");
+    #[rustfmt::skip]
+    qemu.args([
+        "-device", &device,
+        "-kernel", "tock2/target/riscv32imac-unknown-none-elf/release/hifive1",
+        "-M", "sifive_e,revb=true",
+        "-nographic",
+    ]);
+    // QEMU does something to its stdin that prevents Ctrl+C from generating
+    // SIGINT. If we set QEMU's stdin to be our stdin, then Ctrl+C will not
+    // close us. To prevent that, we set QEMU's stdin to null.
+    qemu.stdin(Stdio::null());
```

Comment:
Hmm, I suppose this prevents testing the process console (or any apps which take user input from the console)?

Reply:
Yes. I've now spent over two days trying to get that working without making it difficult to close the runner.

I think there are two separate things you might want to interact with: the emulated UART (for the process console), and QEMU's monitor (its own command line). I think by default, QEMU routes its standard input to its monitor.

I think this is very fixable, either through changing the QEMU flags, or by using `qemu.stdin(Stdin::piped());` and spinning up a thread to forward `runner`'s standard input to QEMU's standard input.

Reply:
OK, sounds good. I don't think we need to block this PR on supporting user input, just wanted to verify there was some path toward supporting it eventually.

Reply:
Alistair may be able to help here. I'm not quite sure what config we want, however -- routing stdin to the UART or to QEMU's monitor. For integration tests we may want to have both, which would involve setting up (and then cleaning up) a UNIX pipe.

Reply:
I added a comment to expose both guest input and monitor to stdin

I don't think we should be connecting stdin to null

Reply:
Okay, after another day of work I have QEMU's stdin working.